src/Avalonia.Styling/Styling/Style.cs

```diff
@@ -112,25 +120,11 @@ public void Attach(IStyleable control, IStyleHost container)
             }
         }
 
-        /// <summary>
-        /// Tries to find a named resource within the style.
-        /// </summary>
-        /// <param name="name">The resource name.</param>
-        /// <returns>
-        /// The resource if found, otherwise <see cref="AvaloniaProperty.UnsetValue"/>.
-        /// </returns>
-        public object FindResource(string name)
+        /// <inheritdoc/>
+        public bool TryGetResource(string key, out object result)
```

Comment:
I think it's a bit confusing to use a method named `FindResource` (with or without Try) both for finding items in a specific node vs. finding them up the hierarchy. WPF made this distinction - Find was for the hierarchy, and you could use the dictionary's indexer or `Resources.TryGetValue` to search a specific node. So I vote to eliminate this method, and use the `Resources` property.

Reply:
Not sure what you mean here, `TryGetResource` calls `_resource.TryGetResource` which searches the resource dictionary and all merged dictionaries.

Reply:
Yes, but it doesn't search the hierarchy (e.g. control > control > app). In WPF, to access the resources of any item in the dictionary, including the merged ones, you use the `control.Resources["MyKey"]`. To search up the hierarchy, you use `control.TryFindResource`.